Missing overload (?)

```cpp
#include <xtensor/xtensor.hpp>
#include <xtensor/xio.hpp>
#include <xtensor/xview.hpp>
#include <xtensor/xfixed.hpp>
#include <xtensor/xadapt.hpp>

using T4 = xt::xtensor_fixed<double, xt::xshape<3,3,3,3>>;

template <class T>
void foo(T&& C)
{
  T4 A;

  C *= A;
}

int main()
{
  xt::xtensor<double, 6> mat = xt::zeros<double>({4,5,3,3,3,3});
  size_t e = 0;
  size_t q = 0;
  auto C = xt::adapt(&mat(e,q,0,0,0,0), xt::xshape<3,3,3,3>());

  foo(C);
}
```

Fails to compile:

```bash
In file included from test.cpp:1:
In file included from /usr/local/include/xtensor/xtensor.hpp:18:
In file included from /usr/local/include/xtensor/xbuffer_adaptor.hpp:20:
In file included from /usr/local/include/xtensor/xstorage.hpp:21:
In file included from /usr/local/include/xtensor/xtensor_simd.hpp:14:
In file included from /usr/local/include/xtensor/xutils.hpp:25:
/usr/local/include/xtl/xsequence.hpp:141:24: error: no matching constructor for initialization of 'xt::xbuffer_adaptor<double *,
      xt::no_ownership, std::__1::allocator<double> >'
                return R(std::begin(r), std::end(r));
                       ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/local/include/xtl/xsequence.hpp:181:27: note: in instantiation of function template specialization
      'xtl::detail::sequence_forwarder_impl<xt::xbuffer_adaptor<double *, xt::no_ownership, std::__1::allocator<double> >,
      xt::aligned_array<double, 81, 8>, void>::forward<xt::aligned_array<double, 81, 8> >' requested here
        return forwarder::forward(std::move(s));
                          ^
/usr/local/include/xtensor/xfixed.hpp:784:26: note: in instantiation of function template specialization
      'xtl::forward_sequence<xt::xbuffer_adaptor<double *, xt::no_ownership, std::__1::allocator<double> >, xt::aligned_array<double, 81, 8>
      &&>' requested here
        m_storage = xtl::forward_sequence<storage_type, origin_type>(std::move(rhs.storage()));
                         ^
/usr/local/include/xtensor/xsemantic.hpp:580:38: note: in instantiation of member function 'xt::xfixed_adaptor<xt::xbuffer_adaptor<double *,
      xt::no_ownership, std::__1::allocator<double> >, xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major,
      xt::xtensor_expression_tag>::operator=' requested here
        return (this->derived_cast() = std::move(tmp));
                                     ^
/usr/local/include/xtensor/xsemantic.hpp:565:37: note: in instantiation of member function
      'xt::xcontainer_semantic<xt::xfixed_adaptor<xt::xbuffer_adaptor<double *, xt::no_ownership, std::__1::allocator<double> >,
      xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major, xt::xtensor_expression_tag> >::assign_temporary' requested here
        return this->derived_cast().assign_temporary(std::move(tmp));
                                    ^
/usr/local/include/xtensor/xsemantic.hpp:376:16: note: in instantiation of function template specialization
      'xt::xsemantic_base<xt::xfixed_adaptor<xt::xbuffer_adaptor<double *, xt::no_ownership, std::__1::allocator<double> >,
      xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major, xt::xtensor_expression_tag>
      >::operator=<xt::xfunction<xt::detail::multiplies, const xt::xfixed_adaptor<xt::xbuffer_adaptor<double *, xt::no_ownership,
      std::__1::allocator<double> >, xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major, xt::xtensor_expression_tag> &, const
      xt::xfixed_container<double, xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major, xt::xtensor_expression_tag> &> >' requested here
        return operator=(this->derived_cast() * e.derived_cast());
               ^
test.cpp:14:5: note: in instantiation of function template specialization 'xt::xsemantic_base<xt::xfixed_adaptor<xt::xbuffer_adaptor<double
      *, xt::no_ownership, std::__1::allocator<double> >, xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major,
      xt::xtensor_expression_tag> >::operator*=<xt::xfixed_container<double, xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major,
      xt::xtensor_expression_tag> >' requested here
  C *= A;
    ^
test.cpp:24:3: note: in instantiation of function template specialization 'foo<xt::xfixed_adaptor<xt::xbuffer_adaptor<double *,
      xt::no_ownership, std::__1::allocator<double> >, xt::fixed_shape<3, 3, 3, 3>, xt::layout_type::row_major, xt::xtensor_expression_tag>
      &>' requested here
  foo(C);
  ^
/usr/local/include/xtensor/xbuffer_adaptor.hpp:263:9: note: candidate constructor template not viable: no known conversion from
      'decltype(__c.end())' (aka 'const double *') to 'xt::xbuffer_adaptor<double *, xt::no_ownership, std::__1::allocator<double>
      >::size_type' (aka 'unsigned long') for 2nd argument; dereference the argument with *
        xbuffer_adaptor(P&& data, size_type size, AT&& alloc = allocator_type());
        ^
/usr/local/include/xtensor/xbuffer_adaptor.hpp:267:9: note: candidate constructor not viable: requires 1 argument, but 2 were provided
        xbuffer_adaptor(const self_type&) = default;
        ^
/usr/local/include/xtensor/xbuffer_adaptor.hpp:270:9: note: candidate constructor not viable: requires 1 argument, but 2 were provided
        xbuffer_adaptor(self_type&&) = default;
        ^
/usr/local/include/xtensor/xbuffer_adaptor.hpp:260:9: note: candidate constructor not viable: requires 0 arguments, but 2 were provided
        xbuffer_adaptor() = default;
        ^
1 error generated.
```

--- a/include/xtensor/xfixed.hpp
+++ b/include/xtensor/xfixed.hpp
@@ -781,7 +781,8 @@ namespace xt
     inline auto xfixed_adaptor<EC, S, L, Tag>::operator=(temporary_type&& rhs) -> self_type&
     {
         using origin_type = decltype(std::move(rhs.storage()));
-        m_storage = xtl::forward_sequence<storage_type, origin_type>(std::move(rhs.storage()));
+        m_storage.resize(rhs.storage().size());
+        std::copy(rhs.storage().cbegin(), rhs.storage().cend(), m_storage.begin());
         return *this;
     }
 

--- a/test/test_xfixed.cpp
+++ b/test/test_xfixed.cpp
@@ -163,6 +163,14 @@ namespace xt
         EXPECT_EQ(a[0], 2);
     }
 
+    TEST(xtensor_fixed, buffer_adaptor)
+    {
+        xtensor_fixed<double, xshape<3>> a;
+        xtensor<double, 2> b = zeros<double>({3, 3});
+        auto c = adapt(&b(0, 0), xshape<3>());
+        c *= a;
+    }
+
     TEST(xtensor_fixed, layout)
     {
         xtensor_fixed<double, xshape<2, 2>, layout_type::row_major> a;

